src/ixdat/plotters/ec_plotter.py

```diff
@@ -0,0 +1,126 @@
+from .plotting_tools import color_axis
+
+
+class ECPlotter:
+    """A matplotlib plotter specialized in electrochemistry measurements."""
+
+    def __init__(self, measurement=None):
+        """Initiate the ECPlotter with its default Meausurement to plot"""
+        self.measurement = measurement
+
+    def plot_measurement(
+        self,
+        *,
+        measurement=None,
+        tspan=None,
+        V_str=None,
+        J_str=None,
+        axes=None,
+        V_color="k",
+        J_color="r",
+        **kwargs,
+    ):
+        """Plot two variables on two y-axes vs time
+
+        All arguments are optional. By default it plots potential in black on the left
+        y-axis and current in red on the right y-axis, using data from its entire
+        measurement. The axes are colored to match the traces and labled witht the
+        respective series names.
+
+        Args:
+            measurement (Measurement): The measurement to plot, if not the one the
+                plotter was initiated with.
+            tspan (iter of float): The timespan (wrt to measurement.tstamp) to plot.
+            V_str (string): The name of the ValueSeries to plot on the left y-axis.
+                Defaults to measurement.V_str, which for an ECMeasurement is the name
+                of its most calibrated/correct potential.
+            J_str (string): The name of the ValueSeries to plot on the right y-axis.
+                Defaults to measurement.J_str, which for an ECMeasurement is the name
+                of its most normalized/correct current.
+            axes (list of matplotlib.Axis): Two axes to plot on, if not the default
+                new twinx()'d axes. axes[0] is for V_str and axes[1] for J_str.
+            V_color (str): The color to plot the series called V_str. Defaults to black.
+            J_color (str): The color to plot the series called J_str. Defaults to red.
+            kwargs (dict): Additional key-word arguments are passed to matplotlib's
+                plot() function, for both potential and current.
+
+        Returns list of matplotlib.pyplot.Axis: The axes plotted on.
+        """
+        measurement = measurement or self.measurement
+        V_str = V_str or (
+            measurement.V_str
+            if measurement.RE_vs_RHE is not None
+            else measurement.E_str
+        )
+        J_str = J_str or (
+            measurement.J_str if measurement.A_el is not None else measurement.I_str
+        )
+        t_v, v = measurement.grab(V_str, tspan=tspan)
+        t_j, j = measurement.grab(J_str, tspan=tspan)
+        if axes:
+            ax1, ax2 = axes
+        else:
+            fig, ax1 = plt.subplots()
+            ax2 = ax1.twinx()
+            axes = [ax1, ax2]
+        ax1.plot(t_v, v, "-", color=V_color, label=V_str, **kwargs)
+        ax2.plot(t_j, j, "-", color=J_color, label=J_str, **kwargs)
+        ax1.set_xlabel("time / [s]")
+        ax1.set_ylabel(V_str)
```

Comment:
`V_str` and `J_str` are renamed `v_name` and `j_name`. 
No option to change axis label added: the user can do that with the axis object returned.

Reply:
Ack on no custom labels.

I found the Napoleon kwargs documentation syntax. It was for spitze_db I found it:
```python
    def ms_scan(self, scan_name="survey", start=None, end="now", SEM=False, **plot_kwargs):
        """Returns scan(s) of specified scan name, e.g. "M4" or "M28-CEM", within a given
        time period. If no scan_name input is given, method return survey(s), i.e. full
        scan(s). If no start input is given, method return latest scan or full survey.

        .. note::
           The data will be saved to and retrieved from the directory cache **only** if both
           ``start`` and ``end`` are present and are absolute time stamp which are in the past.

        Args:
            scan_name (str): Scan name. By default, ``"survey"``, i.e. full scan.
            start (str): Initial time of data extraction. Given as a timestamp i.e.
                ``"YYYY-MM-DD"``, ``"YY-MM-DD hh:mm"`` etc. (see documentation of
                :py:func:`convert_time_to_timestamp`), or time interval
                ``"[]Y[]M[]W[]D[]h[]m"``. By default, None.
            end (str): Final time of data extraction. Format as in start. If no specific time
                of day is given, entire day is included in extracted data. By default, now.
            SEM (bool): Return scan data with (True) or without (False) SEM. By default,
                False.
            **plot_kwargs: See Keyword Arguments below

        Keyword Arguments:
            plot (bool): (optional) Whether to produce a plot instead of returning the data
            title (str): (optional) The plot title
            ylog (bool): (optional) Whether the plot y-axis should use a logartihmic scale
            ymin (float): (optional) Lower cut-off for the y-axis of the plot
            ymax (float): (optional) Upper cut-off for the y-axis of the plot

        Returns:
            sequence of (x, y, metadata_dict): Scans as a pair of numpy arrays (mass, signal)
            with corresponding metadata, i.e. "timestamp", "action_id",
            "ionization_energy", "SEM_voltage", stored in metadata_dict, in sequence
            according to time.

        """
```

renders into:

![billede](https://user-images.githubusercontent.com/1611076/141366497-00180e55-f02c-48e1-a17d-43b2c3e1f710.png)

Reply:
Will unresolve to make sure you see it. It's quite need. The only annoying thing is that you can't rename it, like I would want here, since the name is here plot_kwargs, but alas

Reply:
I've done it for the relevant methods of this plotter. Will leave other relevant places for larger documentation work in the future.